packages/gatsby/src/schema/run-sift.js

```diff
@@ -149,20 +149,20 @@ module.exports = ({
     Object.keys(fieldsToSift).length === 1 &&
     Object.keys(fieldsToSift)[0] === `id`
   ) {
-    const node = resolveRecursive(
+    const nodePromise = resolveRecursive(
       getNode(siftArgs[0].id[`$eq`]),
       fieldsToSift,
       type.getFields()
     )
 
-    if (node) {
+    nodePromise.then(node =>
```

Comment:
we still need to check if `node` actually exist before adding dependency (that promise can resolve with `null` - which is correct)